datatransfer/impl/graphsync/graphsync.go

```diff
@@ -273,17 +277,39 @@ func (receiver *graphsyncReceiver) voucherFromRequest(incoming message.DataTrans
 	return voucher, nil
 }
 
+// ReceiveResponse handles responding to Push or Pull Requests.
+// It schedules a graphsync transfer only if a Pull Request is accepted.
 func (receiver *graphsyncReceiver) ReceiveResponse(
 	ctx context.Context,
 	sender peer.ID,
 	incoming message.DataTransferResponse) {
-		var evt datatransfer.Event
-		if !incoming.Accepted() {
-			evt = datatransfer.Error
-		} else {
-			evt = datatransfer.Progress // for now
+	evt := datatransfer.Error
+	chst := datatransfer.EmptyChannelState
+	if incoming.Accepted() {
+		chid := datatransfer.ChannelID{
+			To: sender,
```

Comment:
non-blocking but make sure you update this when you change To to Initiator. Because the Initiator will be "" - if you are receiving a response, you are the initiator of the request.